store/sqlstore/channel_store_categories.go

```diff
@@ -504,23 +508,7 @@ func (s SqlChannelStore) getSidebarCategoriesT(transaction *gorp.Transaction, us
 }
 
 func (s SqlChannelStore) GetSidebarCategories(userId, teamId string) (*model.OrderedSidebarCategories, error) {
-	transaction, err := s.GetReplica().Begin()
-	if err != nil {
-		return nil, errors.Wrap(err, "begin_transaction")
-	}
-
-	defer finalizeTransaction(transaction)
-
-	oc, err := s.getSidebarCategoriesT(transaction, userId, teamId)
-	if err != nil {
-		return nil, err
-	}
-
-	if err = transaction.Commit(); err != nil {
-		return nil, errors.Wrap(err, "commit_transaction")
-	}
-
-	return oc, nil
+	return s.getSidebarCategoriesT(s.GetReplica(), userId, teamId)
```

Comment:
Let's add a comment this time mentioning the reason for the introduction of the interface, so that the context is preserved.